### PR TITLE
Add modal component to V3

### PR DESCRIPTION
**Pull request by dpikt**
A redo of #426 that doesn't rely on the unmerged `style` branch. Since this is a new component, requiring a style change to use it is not technically breaking 🙂 

## Author Checklist

- [x] Add unit test(s)
- [x] Update version in package.json
- [x] Update documentation (if necessary)
- [x] Add story to storybook (if necessary)
- [x] Assign dev reviewer

### CODE REVIEW

File: src/modal.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+const defaultProps = {
+  hideCloseButton: false,
+}
+
+function getRootElement() {
+  // Skip in SSR mode
+  if (isServer()) return
+  // Note that this expects a root element with id "root"
+  // eslint-disable-next-line no-undef
+  return window.document.querySelector('#root')
```
Comment (chawes13) [inline src/modal.js:56]: Not sure if we discussed this question on the old PR or something else, but any concerns with using `body` here?

Comment (dpikt) [reply src/modal.js:56]: Deja vu! I must've forgotten to copy that commit over. https://github.com/LaunchPadLab/lp-components/pull/426#discussion_r402605702

File: src/modal.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+function Modal({ onClose, hideCloseButton, children, ...rest }) {
+  return (
+    <ReactModal
+      isOpen
+      onRequestClose={onClose}
+      portalClassName="modal"
+      className="modal-inner"
+      overlayClassName="modal-fade-screen"
+      bodyOpenClassName="modal-open"
+      appElement={getRootElement()}
```
Comment (chawes13) [inline src/modal.js:69]: Should this be memoized as a ref?

Comment (dpikt) [reply src/modal.js:69]: Good question- I think the function is basic enough that it's not worth the overhead.

**Pull request merged**